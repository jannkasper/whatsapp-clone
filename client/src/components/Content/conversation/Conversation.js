import React, { useRef, useEffect } from "react";
import styles from "./conversation.module.scss";
import Message from "./message";

function Conversation({ currentConversation, username }) {

    const conversationRef = useRef(null)
    const messagesList = useRef(null);

    useEffect(() => {
        conversationRef.current.scrollTop = messagesList.current.offsetHeight;
    })

    return (
        <div className={styles.conversationContainer}>
            <div className={styles.backgroundContainer}></div>
            <div className={styles.viewSize}>
                <div ref={conversationRef} className={styles.messageGroup}>
                    <div className={styles.conversationEmptySpace}></div>
                    <div ref={messagesList}>
                        {currentConversation?.conversation.map((message, key) => <Message key={key} type={message.type} value={message.value} created={message.created} isAuthor={message.userExtId !== currentConversation.contactExtId}/>)}
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Conversation
