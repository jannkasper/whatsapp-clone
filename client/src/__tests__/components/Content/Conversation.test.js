import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from '../../../store';
import Conversation from '../../../components/Content/conversation';

const initState = {
    conversation: {
        selectedConversation: null
    }
};
const store = configureStore(initState);

describe('Conversation component', () => {

    it('should display', () => {
        const {asFragment} = render(
            <Provider store={store}>
                <Conversation />
            </Provider>,
        );

        expect(asFragment()).toMatchSnapshot();
    })
});
