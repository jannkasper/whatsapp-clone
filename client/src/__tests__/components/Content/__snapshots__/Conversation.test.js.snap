// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conversation component should display 1`] = `
<DocumentFragment>
  <div
    class="conversationContainer"
  >
    <div
      class="backgroundContainer"
    />
    <div
      class="viewSize"
    >
      <div
        class="messageGroup"
      >
        <div
          class="conversationEmptySpace"
        />
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
